util/config/fixpath.py

```diff
@@ -1,62 +1,67 @@
 #!/usr/bin/env python
 
+"""Removes path components that begin with $CHPL_HOME from the given path.
+
+    ./fixpath.py path-value [--shell shell]
+
+Example:
+
+    ./fixpath.py "$PATH" --shell=fish
+
+
+This is used by the setchplenv.* scripts to reduce PATH/MANPATH pollution. It
+may be called in several situations:
+
+1. No Chapel environment settings (new shell)
+2. Same $CHPL_HOME as last time (re-running setchplenv in same dir)
+3. Different $CHPL_HOME (cd ../other-chapel-dir).
+4. $CHPL_HOME is set, but path doesn't include an old one.
+   ($CHPL_HOME was hand-set, now setchplenv is run)
+
+For case 1, just return the existing environment variable.
+
+For case 2, return the environment variable without the components
+that begin with $CHPL_HOME.
+
+For case 3, setchplenv invokes this script before setting the new
+$CHPL_HOME.  We still have the old $CHPL_HOME set, so we can remove
+the old $PATH and $MANPATH entries.  The upshot is we do the same thing
+as in case 2.
+
+Case 4 should also be the same as case 2, but we won't remove any
+components since there should be no components starting with $CHPL_HOME.
+Mentioned only to avoid reintroducing #10196 when this function is modified.
+"""
+
 from __future__ import print_function
 
 from os import getenv
 from sys import argv
 import re
 
-def escape_path(p):
-    if len(argv) >=3 and argv[2] == 'fish':
+import optparse
+
+def escape_path(p, delim):
+    if delim == ' ':
         # Suppress splitting of fish so that the path will be treated as a whole
         return '"{}"'.format(p)
     return p
 
-def main(env='PATH', delim=':'):
+def main(env_val, delim=':'):
```

Comment:
I think I suggested this, but maybe change this to `path_val` since since the script is called `fixpath`